---
title: "Class 14: Genomic Analysis"
date: 11/15/2018
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Asthma SNPs
##Examine the asthma SNPs in the MXL data
```{r}
SNP <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
```

Lets focus on the second column of the genotype info to see the percentage of each SNP
```{r}
genotypes <- round (table(SNP[,2])/nrow(SNP)*100, 2)
```

There are `r genotypes[1]`% AA genotypes in the MXL population

### Interpret base qualities in R
```{r}
library(seqinr)
library(gtools)
phred <- asc( s2c("DDDDCDEDCDDDDBBDDDCC@") ) - 33
phred
```

#Population genetics analysis on SNP
```{r}
expression <- read.table("https://bioboot.github.io/bimm143_F18/class-material/rs8067378_ENSG00000172057.6.txt")
inds.aa <- expression$geno == "A/A"
summary(expression$exp[inds.aa])
inds.ag <- expression$geno == "A/G"
summary(expression$exp[inds.ag])
inds.gg <- expression$geno == "G/G"
summary(expression$exp[inds.gg])
boxplot(exp ~ geno, data = expression)
```

Start from fastq file (the input), do quality control (FASTQC), alignment (mapping) using TopHat, then do Cufflink to count the entries, get the table in R, make a boxplot to show the gene expression is different 